core/unit_test/TestComplex.hpp

```diff
@@ -25,7 +25,13 @@
 #else
 #ifdef __CUDA_ARCH__
 #pragma diagnostic push
-#pragma diag_suppress 20208
+// This line triggers "invalid error number" on Jenkins job "CUDA-11.0-NVCC-RDC".
+// But commenting it triggers the error "'long double' is treated as 'double' in device code".
+// It seems that in https://github.com/kokkos/kokkos/pull/7021/files, the fix
+// was to add at the command line.
+// Here, I just tried to add "-Xcudafe --display_error_number" to get the actual error code and got:
+//  Error #3245-D: 'long double' is treated as 'double' in device code
+#pragma diag_suppress 3245
```

Comment:
@masterleinad I think you worked on this in #7021. Could you please review / test on your side ?

Reply:
Haven't checked yet but could it be that error codes are not the same if using `nv_diag_suppress` *vs* `diag_suppress` ?

Reply:
ping (both of you)

Reply:
I just cross-checked in clean `Docker` containers and got:
```c++
#include "cuda_runtime.h"

/**
 * Compiling with:
 *      ./bin/nvcc_wrapper -Werror test.cpp
 *
 * With @c nvcc 12.6 this generates
 *      test.cpp(5): warning #20208-D: 'long double' is treated as 'double' in device code
 * and can be silenced therefore with
 *      #pragma nv_diag_suppress 20208
 *
 * With @c nvcc 11.0, we need
 *      #pragma diag_suppress 3245
*/
#ifdef __NVCC_DIAG_PRAGMA_SUPPORT__
#pragma nv_diagnostic push
#pragma nv_diag_suppress 20208
#else
#ifdef __CUDA_ARCH__
#pragma diagnostic push
#pragma diag_suppress 3245
#endif
#endif

__device__ void test()
{
    [[maybe_unused]]long double var = 0.5;
}

int main() {}
```

Reply:
So indeed the error codes are different.

Reply:
What does that mean?  Do we need both?

Reply:
It means that to suppress this "long double treated as double in device code" warning, we need 20208 when `__NVCC_DIAG_PRAGMA_SUPPORT__` and 3245 otherwise. More precisely you need 20208 if you use `nv_diag_suppress*` and 3245 if using `diag_suppress`.

Reply:
OK I get it now.  Hadn't expanded the code above.